### ISSUE TITLE
Ignore terragrunt generated files

### ISSUE THREAD

**Issue by mosheavni** [gruntwork-io/terragrunt#1908]
Hi, is there any official way to git-ignore or remove terragrunt generated files before init/apply?

### Motivation

If I generated a file, and then deleted this block, it will stay, and could potentially break the apply process (multiple var declaration, etc...).
Also, since most of the files look more-or-less the same, I don't want to commit them to my repository.

### Potential workaround

Use the before-hook to create a simple bash for-loop to remove any file that has the word `terragrunt` in the first line:

```hcl
terraform {
  before_hook "before_hook" {
    commands = ["init"]
    execute = [
      "bash",
      "-c",
      "for i in *;do head -1 $i | grep -q '^# Generated by Terragrunt.' && echo \"$i is terragrunt\" && rm -f $i;done"
    ]
  }
}
```
Which just doesn't work, the files are gone even after init has finished.
Any ideas?

**Comment by denis256** [comment]
Hi, in other projects I used file name prefixes and git ignore to ignore these files, at least it helped to avoid committing auto-generated files

**Comment by yorinasub17** [comment]
I think we can resolve this by having `generate` generate the files in the working dir (where `terragrunt` invokes `terraform`), instead of the terragrunt dir (where `terragrunt.hcl` is defined). That way, you avoid accidentally committing the generated file, and have a way to always clean the generated file via `--terragrunt-source-update` or wiping the cache.

**Comment by mosheavni** [comment]
> I think we can resolve this by having `generate` generate the files in the working dir (where `terragrunt` invokes `terraform`), instead of the terragrunt dir (where `terragrunt.hcl` is defined). That way, you avoid accidentally committing the generated file, and have a way to always clean the generated file via `--terragrunt-source-update` or wiping the cache.

Good idea, better have this as an option like `generate_locally` with a default of `true`.
Any idea when could this be released?

**Comment by mosheavni** [comment]
Any news here
